blog/2017-09-25-streaming-to-another-database.adoc

```diff
@@ -0,0 +1,222 @@
+* Pre-populated MySQL used in our link:docs/tutorial[tutorial]
+* Empty PostgreSQL
+
+There are certain impedances between Debezium MySQL Connector and Kafka Connect JDBC connector
+
+* event format
+* delete records/tombstone records handling
+* topic naming
+
+We are going to go through all those issues in detail.
+
+=== Event format
+Debezium emits events in a complex format that contains two record values: a value of the row record before the change and value of the row record after the change is applied.
+Further metadata are also included and the result is so called `Envelope` format.
+
+[source,json,indent=0]
+----
+{
+	"op": "u",
+	"source": {
+		...
+	},
+	"ts_ms" : "...",
+	"before" : {
+		"field1" : "oldvalue1",
+		"field2" : "oldvalue2"
+	},
+	"after" : {
+		"field1" : "newvalue1",
+		"field2" : "newvalue2"
+	}
+}
+----
+
+Kafka Connect JDBC connector (as most of the others) emits outgoing and expects incoming events in a simple format that is generally a new row record.
+
+[source,json,indent=0]
+----
+{
+	"field1" : "newvalue1",
+	"field2" : "newvalue2"
+}
+----
+
+Debezium provides a bridge between those two formats in a form of a https://cwiki.apache.org/confluence/display/KAFKA/KIP-66%3A+Single+Message+Transforms+for+Kafka+Connect[single message transform].
+The `UnwrapFromEnvelope` transformation automatically extracts a new row record and thus effectively _flattens_ the complex record into a simple one consumable by other connectors.
+The SMT is applicable at both source and sink connectors but we generally recommend to apply it at sink connectors as then the records stored in Kafka will contain an associated context.
+In our example we apply the SMT at sink connector using configuration properties
+
+[source,indent=0]
+----
+        "transforms": "unwrap", 
+        "transforms.unwrap.type": "io.debezium.transforms.UnwrapFromEnvelope", 
+----
+
+=== Delete records
+When a record is deleted from source database Debezium creates two records: delete record and tombstone record (for compaction).
+Kafka Connect JDBC Sink is not able to work with neither of those to and any attempt to use them leads to `NullPointerException`.
+`UnwrapFromEnvelope` by default filters out both delete and tombstone records and no action is requested on users side.
+If a downstream connector is able to handle those events it is possible to configure `UnwrapFromEnvelope` to pass them downstream.
+
+=== Topic naming
+Last but not least there is a difference in naming of topics.
+Debezium uses fully qualified naming for target topics representing each table it manages.
+The naming follows the pattern `<instance-name>,<database-name>.<table-name>`.
+Kafka Connect JDBC Connector works with simple names `<table-name>`.
+
+In complex scenarios the user would deploy https://kafka.apache.org/documentation/streams/[Kafka Streams] framework to establish elaborated routing between source and target routes.
```

Comment:
> In a more complex pipeline use case, we might use https://kafka.apache.org/documentation/streams/[Kafka Streams] to perform a more complex transformation, aggregation, and/or join of multiple topics, and this same Streams application could also route the messages to the correct topic.
>
> But in our simpler example we can instead use Kafka Connect's built-in `RegexRouter` SMT to send the messages to a different topic.

@jpechane, might be useful to say whether this SMT will be used in the source or sink connector. IIUC, you are using it in the source connector, and if that's right you might add something like this to the previous paragraph: "Again, we could use this SMT in either the source or sink connectors, but for this example we're going to use it in the source connector so we can choose the names of the Kafka topics where the records will be written."